pkg/mpc-examples/contracts/03-ebu-rebalancer/EBURebalancerControllerFactory.sol

```diff
@@ -0,0 +1,139 @@
+            type(EBURebalancerController).creationCode,
+            abi.encode(balancerVault, minimalParams.minSwapFeePercentage)
+        );
+        address expectedControllerAddress = Create2.computeAddress(controllerSalt, keccak256(controllerCreationCode));
+
+        // Build arguments to deploy pool from factory.
+        address[] memory assetManagers = new address[](minimalParams.tokens.length);
+        for (uint256 i = 0; i < assetManagers.length; i++) {
+            assetManagers[i] = expectedControllerAddress;
+        }
+
+        // Populate IManagedPoolFactory.NewPoolParams with arguments from MinimalPoolParams and
+        // other arguments that this factory provides itself.
+        IManagedPoolFactory.NewPoolParams memory fullParams;
+        fullParams.name = minimalParams.name;
+        fullParams.symbol = minimalParams.symbol;
+        fullParams.tokens = minimalParams.tokens;
+        fullParams.normalizedWeights = minimalParams.normalizedWeights;
+        // Asset Managers set to the controller address, not known by deployer until creation.
+        fullParams.assetManagers = assetManagers;
+        fullParams.swapFeePercentage = minimalParams.minSwapFeePercentage;
+        fullParams.swapEnabledOnStart = false;
+        // Factory enforces public LPs for MPs with EBURebalancerController.
+        fullParams.mustAllowlistLPs = false;
+        fullParams.managementAumFeePercentage = minimalParams.managementAumFeePercentage;
+        fullParams.aumFeeId = minimalParams.aumFeeId;
+
+        IManagedPool pool = IManagedPool(
+            IManagedPoolFactory(managedPoolFactory).create(fullParams, expectedControllerAddress)
+        );
+        _lastCreatedPool = address(pool);
+
+        address actualControllerAddress = Create2.deploy(0, controllerSalt, controllerCreationCode);
+        require(expectedControllerAddress == actualControllerAddress, "Deploy failed");
+
+        // Log controller locally.
+        isControllerFromFactory[actualControllerAddress] = true;
+
+        // Log controller publicly.
+        emit ControllerCreated(actualControllerAddress, balancerVault, pool.getPoolId());
+    }
+
+    /**
+     * @dev Allow the owner to disable the factory, preventing future deployments.
+     * @notice owner is initially the factory deployer, but this role can be transferred.
+     * @dev The onlyOwner access control paradigm is an example. Any access control can
+     * be implemented to allow for different needs.
+     */
+    function disable() external onlyOwner {
+        _ensureEnabled();
+        _disabled = true;
+        emit Disabled();
+    }
+
+    function isDisabled() public view returns (bool) {
+        return _disabled;
+    }
+
+    function _ensureEnabled() internal view {
+        require(!isDisabled(), "Controller factory disabled");
+        require(!IManagedPoolFactory(managedPoolFactory).isDisabled(), "Pool factory disabled");
+    }
+}
```

Comment:
Once https://github.com/orbcollective/mpc-examples/pull/7 is approved and merged, let's update these to reflect those changes